pkg/nfd/nodefeaturerule.go

```diff
@@ -0,0 +1,250 @@
+	// Created Builder object on the cluster.
+	Object *nfdv1.NodeFeatureRule
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before Builder object is created.
+	errorMsg string
+}
+
+// NewNodeFeatureRuleBuilderFromObjectString creates a Builder object from CSV alm-examples.
+func NewNodeFeatureRuleBuilderFromObjectString(apiClient *clients.Settings, almExample string) *NodeFeatureRuleBuilder {
+	glog.V(100).Infof(
+		"Initializing new Builder structure from almExample string")
+
+	if apiClient == nil {
+		glog.V(100).Info("The apiClient of the Policy is nil")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(nfdv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Info("Failed to add nfd v1 scheme to client schemes")
+
+		return nil
+	}
+
+	nodeFeatureRule, err := getNodeFeatureRuleFromAlmExample(almExample)
+
+	glog.V(100).Infof(
+		"Initializing Builder definition to NodeFeatureRule object")
+
+	nodeFeatureRuleBuilder := NodeFeatureRuleBuilder{
+		apiClient:  apiClient,
+		Definition: nodeFeatureRule,
+	}
+
+	if err != nil {
+		glog.V(100).Infof(
+			"Error initializing NodeFeatureRule from alm-examples: %s", err.Error())
+
+		nodeFeatureRuleBuilder.errorMsg = fmt.Sprintf("error initializing NodeFeatureRule from alm-examples: %s",
+			err.Error())
+
+		return &nodeFeatureRuleBuilder
+	}
+
+	if nodeFeatureRuleBuilder.Definition == nil {
+		glog.V(100).Infof("The NodeFeatureRule object definition is nil")
+
+		nodeFeatureRuleBuilder.errorMsg = "NodeFeatureRule definition is nil"
+
+		return &nodeFeatureRuleBuilder
+	}
+
+	return &nodeFeatureRuleBuilder
+}
+
+// FeatureRulePull loads an existing NodeFeatureRuleBuilder into Builder struct.
+func FeatureRulePull(apiClient *clients.Settings, name, namespace string) (*NodeFeatureRuleBuilder, error) {
+	glog.V(100).Infof("Pulling existing NodeFeatureRule name: %s in namespace: %s", name, namespace)
+
+	if apiClient == nil {
+		glog.V(100).Info("The apiClient of the NodeFeatureRule is nil")
+
+		return nil, fmt.Errorf("the apiClient of the Policy is nil")
+	}
+
+	err := apiClient.AttachScheme(nfdv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Info("Failed to add nfd v1 scheme to client schemes")
+
+		return nil, err
+	}
+
+	ruleBuilder := NodeFeatureRuleBuilder{
+		apiClient: apiClient,
+		Definition: &nfdv1.NodeFeatureRule{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("NodeFeatureRule name is empty")
+
+		return nil, fmt.Errorf("NodeFeatureRule 'name' cannot be empty")
+	}
+
+	if namespace == "" {
+		glog.V(100).Infof("NodeFeatureRule namespace is empty")
+
+		return nil, fmt.Errorf("NodeFeatureRule 'namespace' cannot be empty")
+	}
+
+	if !ruleBuilder.Exists() {
+		return nil, fmt.Errorf("NodeFeatureRule object %s does not exist in namespace %s", name, namespace)
+	}
+
+	ruleBuilder.Definition = ruleBuilder.Object
+
+	return &ruleBuilder, nil
```

Comment:
see above about returns consistency (#355)
```suggestion
	return ruleBuilder, nil
```